frontend/src/pages/lesson/Insights.jsx

```diff
@@ -1,103 +1,95 @@
-    if (curr.isCompleted) {
-      acc += 1;
-    }
-
-    return acc;
-  }, 0);
 
-  return total !== 0 ? Math.floor((completed / total) * 100) : 100;
-};
-
-const columns = [
-  {
-    field: 'name',
-    headerName: 'Naam',
-    flex: 0.2,
-    valueGetter: (params) => params.row.student.name,
-  },
-  {
-    field: 'progress',
-    headerName: 'Voortgang (%)',
-    flex: 0.1,
-    valueGetter: (params) =>
-      calculateProgress(params.row.questionGroupAttempts),
-  },
-  {
-    field: 'duration',
-    headerName: 'Tijdsduur (minuten)',
-    flex: 0.15,
-  },
-  { field: 'correct', headerName: 'Correct', flex: 0.1, type: 'number' },
-  { field: 'incorrect', headerName: 'Incorrect', flex: 0.1, type: 'number' },
-  { field: 'performance', headerName: 'Prestatie', flex: 0.1, type: 'number' },
-];
+import { observer } from 'mobx-react-lite';
 
-const useStyles = makeStyles((theme) => ({
-  datagrid: {
-    marginTop: '1rem',
-    '& .MuiDataGrid-row:hover': {
-      cursor: 'pointer',
-    },
-  },
-}));
+import InsightsAfter from './InsightsAfter';
+import InsightsDuring from './InsightsDuring';
+import Button from '@material-ui/core/Button';
 
 const Lesson = (props) => {
-  const classes = useStyles();
-  const history = useHistory();
-  const params = useParams();
-
+  const [showCategory, setShowCategory] = useState('after');
   const { lessonStore } = props;
+  const params = useParams();
 
   useEffect(() => {
-    lessonStore.loadLessonAttempts(params.lessonId);
+    lessonStore.loadLesson(params.lessonId);
   }, [lessonStore, params.lessonId]);
 
   if (lessonStore.isLoading) {
     return <CircularProgress />;
   }
+  const changeCategory = () => {
+    setShowCategory(showCategory === 'during' ? 'after' : 'during');
+  };
 
-  const onClickStudent = (event) =>
-    history.push(`/dashboard/students/${event.row.student.id}`);
+  const getQuestionGroupIds = () => {
+    const ids = [];
+    if (lessonStore.lesson) {
+      lessonStore.lesson.questionGroups.forEach((qg) => {
+        ids.push(qg.id);
+      });
+      return ids;
+    }
+  };
+
+  const getContent = () => {
+    return showCategory === 'after' ? (
+      <InsightsAfter {...props} />
+    ) : (
+      <InsightsDuring
+        questionGroupIds={getQuestionGroupIds()}
+        lessonId={params.lessonId}
+        classId={params.classId}
+        {...props}
+      />
+    );
+  };
 
   return (
     <Grid container spacing={2}>
-      <Grid item xs={12} md={6}>
-        <Doughnut
-          lessonAttempts={lessonStore.lessonAttempts}
-          title="Algemene voortgang"
-        />
-      </Grid>
-      <Grid item xs={12} md={6}>
-        <Histogram
-          lessonAttempts={lessonStore.lessonAttempts}
-          title="Cijfer verdeling"
-        />
+      <Grid
+        container
+        spacing={2}
+        style={{ justifyContent: 'center', display: 'flex' }}
+      >
+        <Grid
+          item
+          xs={4}
+          md={4}
```

Comment:
als `xs` op 4 wordt gezet, dan is `md` ook automatisch 4 (ook lg, xl for that matter) dus die kan weg :)